cli_tests/app.sh

```diff
@@ -0,0 +1,172 @@
+  [[ $result =~ "For details see https://theketch.io" ]]
+  [[ $result =~ "Available Commands" ]]
+  [[ $result =~ "Flags" ]]
+}
+
+@test "framework add" {
+  result=$($KETCH framework add "$FRAMEWORK" --ingress-service-endpoint "$INGRESS" --ingress-type "traefik")
+  echo "RECEIVED:" $result
+  [[ $result =~ "Successfully added!" ]]
+}
+
+@test "framework list" {
+  result=$($KETCH framework list)
+  headerRegex="NAME[ \t]+STATUS[ \t]+NAMESPACE[ \t]+INGRESS TYPE[ \t]+INGRESS CLASS NAME[ \t]+CLUSTER ISSUER[ \t]+APPS"
+  dataRegex="$FRAMEWORK[ \t]+ketch-$FRAMEWORK[ \t]+traefik[ \t]+traefik"
+  echo "RECEIVED:" $result
+  [[ $result =~ $headerRegex ]]
+  [[ $result =~ $dataRegex ]]
+}
+
+@test "framework update" {
+  result=$($KETCH framework update "$FRAMEWORK" --app-quota-limit 1)
+  echo "RECEIVED:" $result
+  [[ $result =~ "Successfully updated!" ]]
+}
+
+@test "app deploy" {
+  run $KETCH app deploy "$APP_NAME" --framework "$FRAMEWORK" -i "$APP_IMAGE"
+  [ $status -eq 0 ]
+}
+
+@test "app list" {
+  result=$($KETCH app list)
+  headerRegex="NAME[ \t]+FRAMEWORK[ \t]+STATE[ \t]+ADDRESSES[ \t]+BUILDER[ \t]+DESCRIPTION"
+  dataRegex="$APP_NAME[ \t]+$FRAMEWORK[ \t]+(created|running)"
+  echo "RECEIVED:" $result
+  [[ $result =~ $headerRegex ]]
+  [[ $result =~ $dataRegex ]]
+}
+
+@test "app info" {
+  result=$($KETCH app info "$APP_NAME")
+  headerRegex="DEPLOYMENT VERSION[ \t]+IMAGE[ \t]+PROCESS NAME[ \t]+WEIGHT[ \t]+STATE[ \t]+CMD"
+  dataRegex="1[ \t]+docker.io/shipasoftware/bulletinboard:1.0[ \t]+web[ \t]+100%[ \t]+created[ \t]+docker-entrypoint.sh npm start"
+  echo "RECEIVED:" $result
+  [[ $result =~ $headerRegex ]]
+  [[ $result =~ $dataRegex ]]
+}
+
+@test "app stop" {
+  result=$($KETCH app stop "$APP_NAME")
+  echo "RECEIVED:" $result
+  [[ $result =~ "Successfully stopped!" ]]
+}
+
+@test "app start" {
+  result=$($KETCH app start "$APP_NAME")
+  echo "RECEIVED:" $result
+  [[ $result =~ "Successfully started!" ]]
+}
+
+@test "app log" {
+  run $KETCH app log "$APP_NAME"
+  [ $status -eq 0 ]
+}
+
+@test "builder list" {
+  result=$($KETCH builder list)
+  headerRegex="VENDOR[ \t]+IMAGE[ \t]+DESCRIPTION"
+  dataRegex="Google:[ \t]+gcr.io/buildpacks/builder:v1[ \t]+GCP Builder for all runtimes"
+  echo "RECEIVED:" $result
+  [[ $result =~ $headerRegex ]]
+  [[ $result =~ $dataRegex ]]
+}
+
+@test "cname add" {
+  run $KETCH cname add "$CNAME" --app "$APP_NAME"
+  [ $status -eq 0 ]
+  result=$($KETCH app info "$APP_NAME")
+  echo "RECEIVED:" $result
+  [[ $result =~ "Address: http://$CNAME" ]]
+}
+
+@test "cname remove" {
+  run $KETCH cname remove "$CNAME" --app "$APP_NAME"
+  [ $status -eq 0 ]
+  result=$($KETCH app info "$APP_NAME")
+  echo "RECEIVED:" $result
+  [[ ! $result =~ "Address: http://$CNAME" ]]
+}
+
+@test "unit add" {
+ run $KETCH unit add 1 --app "$APP_NAME"
+ [ $status -eq 0 ]
+ result=$(kubectl describe apps $APP_NAME)
+ echo "RECEIVED:" $result
+ [[ $result =~ "Units:  2" ]] # note two spaces
```

Comment:
Thank you for these comments, the double spaces look like a huge source of frustration for debugging if you don't know about them.

Reply:
The double brackets were required for the regex comparisons: `=~`. I made the `status` equality checks double too, for consistency. Note: I'm not a bash expert.